src/extensionApi.ts

```diff
@@ -81,6 +81,71 @@ export class CommandHandler {
         }
     }
 
+    public async debugServer(context?: Protocol.ServerState): Promise<Protocol.StartServerResponse> {
+
+        const debugInfo = await this.serversData.retrieveDebugInfo(context.server);
+
+        if (debugInfo && debugInfo.properties['debug.details.type'].indexOf('java') >= 0) {
+            if (vscode.extensions.getExtension('vscjava.vscode-java-debug') === undefined) {
+                vscode.window.showErrorMessage('Debugger for Java extension is required. Install/Enable it before proceeding.');
+                return;
+            }
+        } else {
+            vscode.window.showErrorMessage(`Vscode-Adapters doesn\'t support debugging with ${debugInfo.properties['debug.details.type']} language at this time.`);
+            return;
+        }
+
+        // list all projects that belong to the current workspace allowing the user choose among them or pick a new one
+        let optionsQuickPick: Array<{label: string, description: string}>  = [{ label: 'Add a new project', description: undefined}];
+        if (vscode.workspace.workspaceFolders !== undefined) {
+            optionsQuickPick = optionsQuickPick.concat(vscode.workspace.workspaceFolders.map(x => ({label: x.name, description: x.uri.toString()})));
+        }
+
+        let folderProject: vscode.Uri = await vscode.window.showQuickPick(optionsQuickPick,
+                                                { placeHolder: 'Select the project you want to debug', ignoreFocusOut: true }
+                                        ).then(folder => {
+                                            return (!folder || folder.description === undefined ? undefined : vscode.Uri.parse(folder.description));
+                                        });
+
+        if (!folderProject) {
```

Comment:
If the user escape the folders quickpick then by convention it means that user wants to cancel the whole operation. So we need a way to distinguish the 2 cases:

- escape
- user selected `Add new project`